File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/docker/DockerPortBinding.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.docker;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableDockerPortBinding.class)
public interface DockerPortBinding {

  Optional<Integer> hostPort();

  Optional<Integer> containerPort();

  Optional<DockerPortBindingProtocol> protocol();

  Optional<String> hostIp();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/docker/DockerPortBinding.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.docker

data class DockerPortBinding(

    val hostPort: Int? = null,

    val containerPort: Int? = null,

    val protocol: DockerPortBindingProtocol? = null,

    val hostIp: String? = null
)
